crates/apub/src/objects/community.rs

```diff
@@ -150,6 +151,22 @@ impl Object for ApubCommunity {
     } else {
       CommunityVisibility::Public
     });
+
+    // If NSFW is not allowed, reject new communities marked NSFW and
+    // remove communities that update to be NSFW
+    let nsfw_disallowed = local_site.is_some_and(|s| s.disallow_nsfw_content);
```

Comment:
It would be better to use the same function you created. You might need to make `local_site` into an `&Option<LocalSite>`, like I'm seeing a few of the other functions in api_common/utils

Side point but @Nutomic I'm seeing this `let local_site_data = local_site_data_cached(&mut context.pool()).await?` in some places, and other places it runs `LocalSite::read(...`

Might be good to make an issue about going over all those in the apub code, and making them all use the cached version.

Reply:
LocalSite::read is also cached so thats fine. However I found that slur filter handling is unnecessarily complicated, will open a PR shortly.